pyOCD/pyDAPAccess/analyzer/cmsis_dap_analyzer.py

```diff
@@ -0,0 +1,464 @@
+, 'DHCSR':
+    {
+        0x02000000: ('S_RESET_ST',   25)
+      , 0x01000000: ('S_RETIRE_ST',  24) 
+      , 0x00080000: ('S_LOCKUP',     19)
+      , 0x00040000: ('S_SLEEP',      18)
+      , 0x00020000: ('S_HALT',       17)
+      , 0x00010000: ('S_REGRDY',     16)
+      , 0x00000020: ('C_SNAPSTALL',   5)
+      , 0x00000008: ('C_MASKINTS',    3)
+      , 0x00000004: ('C_STEP',        2)
+      , 0x00000002: ('C_HALT',        1)
+      , 0x00000001: ('C_DEBUGEN',     0)
+    }
+, 'AIRCR': 
+    {   
+        0x00008000: ('ENDIANNESS',   15)
+      , 0x00000700: ('PRIGROUP',      8)
+      , 0x00000004: ('SYSRESETREQ',   2)
+      , 0x00000002: ('VECTCLRACTIVE', 1) 
+      , 0x00000001: ('VECTRESET',     0)
+    }
+}
+
+
+RESP_STATUS = {  0xff: 'DAP_ERROR'
+               , 0x00: 'DAP_OK'}
+CONN_RESP = {0x0: 'FAILED', 0x1: 'SWD', 0x2: 'JTAG'}
+
+invmap = lambda d: {v:k for k,v in d.iteritems()}
+AP_REG = invmap(AP_REG)
+COMMAND_ID = invmap(COMMAND_ID)
+ID_INFO = invmap(ID_INFO)
+
+lookup_code = lambda d,k: d.get(k, "UNKNOWN_CODE(0x%02x)" % k)
+lookup_cmd = lambda cmd: COMMAND_ID.get(cmd, "UNKNOWN_CMD(0x%02x)" % cmd)
+reverse_bits = lambda b,width=0: bin(b)[2:].zfill(width)[::-1]
+
+
+def get_word(data,index=0,pop=False):
+    d1,d2,d3,d4 = data[index:index+4]
+    if pop: del data[index:index+4]
+    return d4 << 24 | d3 << 16 | d2 << 8 | d1
+
+class CMSIS_DAPAnalyzer(TransportAnalyzer):
+    def __init__(self, interface):
+        super(CMSIS_DAPAnalyzer, self).__init__(interface)
+        self.last_rqst = deque()
+        self.last_blktransfer = []
+        self._cache = {}
+
+    def default_trace(self, data):
+        ncols = 16
+        nlines, r = divmod(len(data),ncols)
+        
+        lines = ''
+        if nlines > 0: 
+            lines += '\n\t' + '\n\t'.join([' '.join(['0x{:02x}']*ncols)]*nlines)
+        if r > 0: 
+            lines += '\n\t' + ' '.join(['0x{:02x}']*r)
+
+        return lines.format(*data)
+
+    def get_rqst_translater(self, cmd):
+        return getattr(self, cmd.lower()+'_rqst', None)
+
+    def get_resp_translater(self, cmd):
+        return getattr(self, cmd.lower()+'_resp', None)
+
+    def trace_write(self, data, **kwargs):
+        cmd = lookup_cmd(data[0])
+        if cmd == 'DAP_TRANSFER':
+            (dapidx, tcnt), xfers = data[1:3], data[3:]
+
+            header = " (dap_idx=0x{:02x}, xfer_cnt=0x{:02x})".format(dapidx,tcnt)
+            trace = header + self._xrqst(tcnt, xfers)
+        elif cmd == 'DAP_TRANSFER_BLOCK':
+            dapidx, (tcntL, tcntH), treq, xfers = data[1], data[2:4], data[4], data[5:]
+            tcnt = tcntH << 8 | tcntL
+
+            header = " (dap_idx=0x{:02x}, xfer_cnt=0x{:02x})".format(dapidx,tcnt)
+            trace = header + self._xrqst_block(tcnt, treq, xfers)
+        else:
+            tracer = self.get_rqst_translater(cmd)
+            trace = tracer(data) if tracer is not None else ""
+
+        # logging.debug(("trace_tx:%s" % cmd.lower()) + trace)
```

Comment:
@c1728p9 Mostly to reduce clutter. The transmit messages are very much similar to the receive messages, except they doesn't include any actual read data or whether the transmission succeeded, so they are pretty much redundant. What would be useful however is to store these logging messages and use them as a sort of partial traceback for a failed transmission.